### ISSUE TITLE
Old fleet-agent is not deleted after rancher server upgrade

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#33677]
**Rancher Server Setup**
- Rancher version: 2.5.8 to master-head

**To Reproduce**
1. Installed a 2.5.8 cluster from rancher-stable with helm:

```
$ helm install rancher rancher-stable/rancher --set hostname=429604a2fec0.ngrok.io --namespace=cattle-system --version=v2.5.8
```

2. Registered a downstream "Other cluster"
3. fleet-agent 0.3.5 is installed in the fleet-system namespace
![image](https://user-images.githubusercontent.com/1220710/125887671-64f4d034-4f54-4b7e-bcc6-7555d8da0dce.png)

4. Built rancher chart from 1b81a0615ddef614e41c8ca53ccb309c447ac350 and upgraded to it:

```
helm upgrade rancher ~/dev/rancher/build/chart/rancher --namespace=cattle-system  --set hostname=429604a2fec0.ngrok.io --set rancherImageTag=master-head
```
5. fleet-agent on downstream cluster is installed in the cattle-fleet-system namespace shortly after cattle-cluster-agent finishes upgrading
![image](https://user-images.githubusercontent.com/1220710/125888206-088e7348-81bd-406a-842b-d03e85061529.png)

The old one is still there, which could be a problem, but the new one is there too. 

**Expected Result**
The old fleet should get deleted

Issue found here https://github.com/rancher/rancher/issues/32560#issuecomment-881153122

**Note:** 
On a recent upgrade from 2.5.9 to master-head commit id: `d3ad5d3`, i dont see this issue. New fleet agent is deployed and Old fleet agent is removed. New fleet agent - `rancher/fleet-agent:v0.3.6-rc5`

**Comment by nickgerace** [comment]
This is no longer seen on `master-head` from upgrades from single-node v2.5.8 and v2.5.9.

**Comment by nickgerace** [comment]
Spoke to @sowmyav27: this was not seen on the two aforementioned different setups. Since the issue could have been caused by changes to Rancher or its related components (i.e. wrangler), I recommend that we close this issue.

**Comment by sowmyav27** [comment]
Closing this, will log a new one or reopen if we see it again

**Comment by sowmyav27** [comment]
Reopening this.
**See this on an upgrade from 2.5.9 to master-head commit id: `cd623b`**

- HA setup
- 2 downstream clusters - AWS node driver - 1etcd/control and 3 worker nodes and 1 DO node driver cluster - 1 etcd/cp and 2 worker nodes.

**Comment by sowmyav27** [comment]
**Validated on an upgrade from 2.5.9 to master-head commit id: `21ab02c`**

- On a docker install of rancher, deploy a downstream cluster.
- Upgrade rancher to master-head
- In the local cluster --> System project --> Noticed Old fleet agent gets deleted first and the new fleet agent is available `rancher/fleet-agent:v0.3.6-rc7`
